include/flecs.h

```diff
@@ -2362,7 +2367,43 @@ void _ecs_assert(
 
 /** Calculate offset from address */
 #define ECS_OFFSET(o, offset) (void*)(((uintptr_t)(o)) + ((uintptr_t)(offset)))
+#define is_empty(...) ( sizeof( (char[]){#__VA_ARGS__} ) == 1 )
 
+const char* if_encapsulation(const char * component_id, const char* signature, uint *position) {
+
+    if(component_id){
+        if(strlen(component_id) == 0){
+            *position = strlen(signature) - 1;
+        } else {
+            char* pointer_to_string = strstr(signature, component_id);
+            *position = pointer_to_string - signature;
+        }
+        return component_id;
+    }
+
+    return signature;
+}
+
+#define ecs_print_error_string(signature, system_id, error_description, component_id, ...)\
```

Comment:
No need to make this a macro, you can change thxs to a function with a variable argument list:

```c
void ecs_print_error_string(signature, system_id, error_description, component_id, ...)
```

Reply:
removed